Dbms milestone 4

ferrary7 opened the pull request:
This is the walkthrough video submission of my DBMS milestone 4, please find the video below:
[VIDEO](https://vimeo.com/881016526/2f1764cc7b?share=copy)

Concepts covered in this self-viva: 
- Joins and Subqueries
- RBAC

Thanks.

ferrary7 committed:
Joins and SubQueries added

ferrary7 committed:
Role based Access conrtol added

PRASANNAPATHI commented:
Hey @ferrary7 ,

For Advanced SQL queries - your explanation was good, but you need to submit atleast 3 queries with explanation, you have submitted only 2.
Please, you can merge it (Advanced SQL queries). If you want  to score fully then please resubmit with things mentioned above.


For RBAC - your idea to implement roles system is appreciable, but it is not what expected, Please go through the RBAC concept in DBMS livebooks, from which you can get an idea of what to do.

Please book a viva slot using the link shared in wiki, after you done with this.

ferrary7 committed:
Updated the RBAC

ferrary7 commented:
Hi @PRASANNAPATHI, I've implemented the concepts as per the remarks given. Below is the gist of it:
- For the Advance SQL queries, I already used 3 before (just didn't point out one). I've commented the required part of code for it.
![image](https://github.com/kalviumcommunity/kalvium_budget_tracker/assets/106926421/7d698796-5932-4d33-bec9-31298a81760a)


- I've replaced the logic earlier with new logic for the RBAC according to the livebooks.

PRASANNAPATHI commented:
Yes @ferrary7 , Everything is fine now,

Please merge it for both Advanced SQL queries and RBAC.
